### ISSUE TITLE
Builder subclass for a generic type generates an invalid declaration

### ISSUE THREAD

**Issue by jfancher** [immutables/immutables#377]
This code:
````java
@Value.Immutable
interface Bug<T> {
    interface Builder<T> {}
}
````

Generates this code:
````java
final class ImmutableBug<T> implements Bug<T> {
    static final class Builder<T> implements Bug<T>.Builder<T> {}
}
````

This fails to compile with:
````
error: cannot select a static class from a parameterized type
````
The derived builder declaration should be `implements Bug.Builder<T>`.

**Comment by elucash** [comment]
Thank you for reporting this! Fixed this one and added additional validation for generic parameters to match. Will be available in the next release (or you can grab snapshots/build from source).

**Comment by elucash** [comment]
The fix released in 2.2.9 Thank you!
